### PR TITLE
Student Profile Page

### LINKED ISSUE #29: Profile Page
Create the profile Page using the provided mocks.

This issue is just for the scaffolding of the profile Page, as according to what was discussed in a previous meeting. The content should be done separately.

### PR DESCRIPTION
**Pull request by pedronunes19**
Closes #29

### CODE REVIEW

**Comment by pedronunes19** [comment]
Very early draft. Student info (name, socials, profile picture, CV) still needs to be properly displayed on this page. Student's will be able to upload files and edit personal info (modal).

**Comment by pedronunes19** [comment]
The profile picture request is returning a 404, although the picture is correctly stored locally.

**Comment by Naapperas** [comment]
> The profile picture request is returning a 404, although the picture is correctly stored locally.

Have you run `php artisan storage:link`?

**Comment by ttoino** [comment]
> > The profile picture request is returning a 404, although the picture is correctly stored locally.
> 
> Have you run `php artisan storage:link`?

Additionally, the `APP_URL` env variable is wrong on `main`, but it's fixed in #69.

**Comment by pedronunes19** [comment]
> > The profile picture request is returning a 404, although the picture is correctly stored locally.
> 
> Have you run `php artisan storage:link`?

I hadn't, thank you

File: resources/js/Components/Profile/CvArea.vue [hunk 1]
```diff
@@ -59,10 +59,23 @@ const clearCVFileInput = () => {
...
+                    >
+                        <path
+                            fill="none"
+                            stroke="currentColor"
+                            stroke-linecap="round"
+                            stroke-linejoin="round"
+                            stroke-width="36"
+                            d="M56 262 v150 h400 M456 262 v150 M256 312 v-250 l-80 80 M256 62 l80 80"
+                        />
+                    </svg>
```
Comment (ttoino) [inline resources/js/Components/Profile/CvArea.vue]: Doing this will no longer be necessary after #81

Comment (pedronunes19) [reply resources/js/Components/Profile/CvArea.vue]: :boom:

File: app/Http/Controllers/CVController.php [hunk 2]
```diff
@@ -14,8 +15,20 @@ class CVController extends Controller
      */
     public function destroy(Request $request)
     {
-        $request->user()->deleteCV();
+        Participant::where('user_id', $request->user()->id)->first()->deleteCV();
```
Comment (Naapperas) [inline app/Http/Controllers/CVController.php:18]: Ideally this should use the `usertype` attribute.

File: app/Models/VisitHistory.php
Comment (Naapperas) [inline app/Models/VisitHistory.php:1]: With the cock

File: app/Providers/AuthServiceProvider.php [hunk 3]
```diff
@@ -25,5 +28,6 @@ public function boot(): void
         Gate::define('admin', fn ($user) => $user->isAdmin());
         Gate::define('participant', fn ($user) => $user->isParticipant());
         Gate::define('company', fn ($user) => $user->isCompany());
+        Gate::define('view_participant', fn ($user, Participant $participant) => ($user->id == $participant->user_id) || (($user->usertype_type == Company::class) && VisitHistory::where('company_id', $user->usertype_id)->where('participant_id', $participant->id)->exists()));
```
Comment (Naapperas) [inline app/Providers/AuthServiceProvider.php]: Cock this.

File: resources/js/Components/Profile/CvArea.vue [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    cvInput.value?.click();
+};
+
+const togglePreview = () => {
+    previewOpen.value = !previewOpen.value;
+};
+
+const uploadCV = () => {
+    const cv = cvInput.value?.files?.[0];
+    console.log(cvInput.value?.files?.[0]);
```
Comment (Naapperas) [inline resources/js/Components/Profile/CvArea.vue]: BRuh

File: resources/js/Components/Profile/CvArea.vue [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    if (cv) {
+        form.cv = cvInput.value?.files?.[0] ?? null;
+        form.post(route("current-user-cv.update"), {
+            preserveScroll: true,
+            onSuccess: () => clearCVFileInput,
```
Comment (Naapperas) [inline resources/js/Components/Profile/CvArea.vue]: Fix

File: resources/js/Pages/Profile/Show.vue [hunk 6]
```diff
@@ -1,73 +1,63 @@
...
 
 defineProps<Props>();
 </script>
 
 <template>
     <AppLayout title="Profile">
-        <template #header>
-            <h2
-                class="text-xl font-semibold leading-tight text-gray-800 dark:text-gray-200"
+        <main class="flex flex-col items-center bg-2023-bg pt-6 sm:pt-0">
```
Comment (Naapperas) [inline resources/js/Pages/Profile/Show.vue]: template

**Review by Naapperas — CHANGES_REQUESTED** [review]
Sorry

File: app/Http/Controllers/CVController.php (hunk 2 above)
Comment (pedronunes19) [reply app/Http/Controllers/CVController.php:18]: I know

File: resources/js/Components/Profile/CvArea.vue (hunk 5 above)
Comment (pedronunes19) [reply resources/js/Components/Profile/CvArea.vue]: fixed

File: resources/js/Pages/Profile/Show.vue (hunk 6 above)
Comment (pedronunes19) [reply resources/js/Pages/Profile/Show.vue]: div

**Review by toni-santos — APPROVED** [review]

**Review by Naapperas — APPROVED** [review]

**Pull request merged**